### PR TITLE
Convert Jade Loader to ES6

### PR DESCRIPTION
**Pull request by elvingm**
Accidentally opened my PR to the official `webpack/jade-loader` :sweat_smile:

### CODE REVIEW

**Comment by kylemac** [comment]
this ready for review @elvingm? i wasn't sure because it wasn't assigned to anyone

**Comment by elvingm** [comment]
@kylemac yep! it is.

**Comment by kylemac** [comment]
we're still using `var` in a bunch of places. let's use `let` if the variable is later updated or `const` if it's not. once you've done that, ping me and i'll do the full review.

**Comment by elvingm** [comment]
@kylemac Updated. At first used `let` mostly, but changed some variables to `const`. Grouped them when declared too

**Comment by kylemac** [comment]
@elvingm lets get those CI tests to pass :tennis:

File: package.json [hunk 1]
```diff
@@ -30,7 +30,7 @@
     "url": "git://github.com/carrot/jade-loader.git"
   },
   "scripts": {
-    "test": "mocha -R spec",
+    "test": "mocha --require babel-core/register",
```
Comment (kylemac) [inline package.json:33]: you still need to update the tests to load `../src` instead of `../lib`

Comment (elvingm) [inline package.json:33]: Ah! That's right. All done now.

**Comment by kylemac** [comment]
LGTM :+1: merge this up and then please rebase the new `master` under #4 and then get #4 merged too

**Comment by elvingm** [comment]
:+1:

**Pull request merged**